src/uhi/testing/indexing.py

```diff
@@ -0,0 +1,515 @@
+from __future__ import annotations
+
+import abc
+import typing
+import unittest
+from typing import Any
+
+import numpy as np
+
+import uhi.tag
+
+T = typing.TypeVar("T", bound=Any)
+
+__all__ = ["Indexing"]
+
+
+if typing.TYPE_CHECKING:
+    # This is a workaround for the issue with type checkers
+    # and generics with custom __getitem__ methods.
+    sum = 42
+
+
+def __dir__() -> list[str]:
+    return __all__
+
+
+class Indexing(typing.Generic[T], abc.ABC, unittest.TestCase):
+    """
+    This test requires histograms to be created first.
+
+    h1 and h3 are histograms created in the test suite.
+    h1 is a 1D histogram with 10 bins from 0 to 1. Each bin has 2 more than the
+    one before, starting with 0. The overflow bin has 1. The underflow bin has 3.
+
+    h3 is a 3D histogram with [2,5,10] bins. The contents are x+2y+3z, where x,
+    y, and z are the bin indices.
+    """
+
+    h1: T
+    h3: T
+    tag = uhi.tag
+
+    @staticmethod
+    @abc.abstractmethod
+    def make_histogram_1() -> T:
+        pass
+
+    @staticmethod
+    @abc.abstractmethod
+    def make_histogram_3() -> T:
+        pass
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        cls.h1 = cls.make_histogram_1()
+        cls.h3 = cls.make_histogram_3()
+
+    def test_access_integer_1d(self) -> None:
```

Comment:
Currently this directly compares the values (here int).
```
tests/uhi/generic.py::TestAccess::test_access_integer_1d - AssertionError: GROGU_HISTO1D_V3.Bin(d_sumw=0.0, d_sumw2=[44 chars]=1.0) != 0
```
In my case the bin also tracks the sumw2 as a members, would it make sense to wrap the `float(self.h1[1])` and my Histo1d bin would default to returning sumw for casting to float?